### ISSUE TITLE
ReferenceError: requestAnimationFrame is not defined

### ISSUE THREAD

**Issue by SpidyBoYz** [vidstack/player#971]
### Current Behavior:

ReferenceError: requestAnimationFrame is not defined

### Steps To Reproduce:

Use Next.JS

### Environment:

- Framework: React
- Meta Framework: Next.js
- Node: 18.10.0
- Device: Windows 11
- OS: Windows
- Browser: Chrome@118.0.5993.89

**Comment by mihar-22** [comment]
Need reproduction. I just tested quickly and there was no issues.

**Comment by AyushAgnihotri2025** [comment]
In Node.js environments (like when static rendering with Expo Router). requestAnimationFrame is unavailable. Do not call side effects in this case. I seems to be fixed, with the below changes.

```diff
  private _attachMenuItems(items: MenuItems) {
    const el = items.el!;
    el.style.setProperty('display', 'none');

    setAttribute(el, 'id', this._menuId);
    setAttributeIfEmpty(el, 'role', 'menu');
    setAttributeIfEmpty(el, 'tabindex', '-1');
    setAttribute(el, 'data-submenu', this.isSubmenu);

    this._content.set(el);
    onDispose(() => this._content.set(null));

    if (!this.isSubmenu) {
      this._stopClickPropagation(el);
    }

    const watchAttrs = () => {
      setAttribute(el, 'data-open', this._expanded());
    };

    if (__SERVER__) watchAttrs();
    else effect(watchAttrs);

    this._focus._attachMenu(el);
    this._updateMenuItemsHidden(false);
+   if (typeof requestAnimationFrame === 'undefined') {
+     return;
+   }
    requestAnimationFrame(this._onResize.bind(this));
  }
```

https://github.com/vidstack/player/blob/7bd452ab38cce20c2cd5e4270c91b0464fa3243a/packages/vidstack/src/components/ui/menu/menu.ts#L224-L250

**Comment by mihar-22** [comment]
Nice catch @AyushAgnihotri2025! Thank you. I quickly patche and released in `1.4.4`. Let me know if the issue persists.
